crates/astria-conductor/src/celestia/mod.rs

```diff
@@ -590,6 +599,40 @@ async fn process_sequencer_blob(
     })
 }
 
+/// Get the sequencer namespace from the latest sequencer block.
+async fn get_sequencer_namespace(client: SequencerClient) -> eyre::Result<Namespace> {
+    use sequencer_client::Client as _;
+
+    let retry_config = tryhard::RetryFutureConfig::new(10)
+        .exponential_backoff(Duration::from_millis(100))
+        .max_delay(Duration::from_secs(20))
+        .on_retry(
+            |attempt: u32,
+             next_delay: Option<Duration>,
+             error: &sequencer_client::tendermint_rpc::Error| {
+                let wait_duration = next_delay
+                    .map(humantime::format_duration)
+                    .map(tracing::field::display);
+                warn!(
+                    attempt,
+                    wait_duration,
+                    error = error as &dyn std::error::Error,
+                    "attempt to grab sequencer block failed; retrying after backoff",
+                );
+                futures::future::ready(())
+            },
+        );
+
+    let response = tryhard::retry_fn(|| client.latest_commit())
```

Comment:
Getting the latest sequencer block was wasteful (we just grab the header after all and throw everything else away).

Kinda figured this out because it felt weird having to mount a mocked block just for this call.

Actutally, I think this one might be even better: https://docs.rs/tendermint-rpc/0.34.1/tendermint_rpc/endpoint/genesis/index.html

Reply:
I am going to move this diff out of this PR. I initially set out to also mock firm-only mode, which is what this change was for.